code/API_definitions/BYON-Webrtc-Events.yaml

```
@@ -0,0 +1,1520 @@
+                        $ref: "#/components/examples/NEW_SESSION_INCOMING"
+                      REGISTRATION_REVOKED:
+                        $ref: "#/components/examples/REGISTRATION_REVOKED"
+                      SESSION_UPDATE:
+                        $ref: "#/components/examples/SESSION_UPDATE"
+              responses:
+                "204":
+                  description: Successful notification
+                  headers:
+                    x-correlator:
+                      $ref: "#/components/headers/x-correlator"
+                "400":
+                  $ref: "#/components/responses/Generic400"
+                "401":
+                  $ref: "#/components/responses/Generic401"
+                "403":
+                  $ref: "#/components/responses/Generic403"
+                "410":
+                  $ref: "#/components/responses/Generic410"
+                "429":
+                  $ref: "#/components/responses/Generic429"
+                "500":
+                  $ref: "#/components/responses/Generic500"
+                "503":
+                  $ref: "#/components/responses/Generic503"
+              security:
+                - {}
+                - notificationsBearerAuth: []
+
+      responses:
+        "201":
+          description: Created
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Subscription"
+        "202":
+          description: Request accepted to be processed. It applies for async creation process.
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SubscriptionAsync"
+        "400":
+          $ref: "#/components/responses/CreateSubscriptionBadRequest400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/SubscriptionPermissionDenied403"
+        "409":
+          $ref: "#/components/responses/Generic409"
+        "415":
+          $ref: "#/components/responses/Generic415"
+        "422":
+          $ref: "#/components/responses/CreateSubscriptionUnprocessableEntity422"
+        "429":
+          $ref: "#/components/responses/Generic429"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+    get:
+      tags:
+        - webrtc-events Subscription
+      summary: "Retrieve a list of webrtc-events event subscription"
+      description: Retrieve a list of webrtc-events event subscription(s)
+      operationId: retrieveNotificationChannelSubscriptionList
+      parameters:
+        - $ref: "#/components/parameters/x-correlator"
+      security:
+        - openId:
+            - api-name:read
```

https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#1161-scope-naming
```suggestion
            - org.camaraproject.webrtc-events.session:read
```

---

In fact, it should include one of each, isn't it? I mean, if I'm not wrong, to have a valid GET reponse you need one of these permissions, one for each type of subscription.
- org.camaraproject.webrtc-events.session-status:read
- org.camaraproject.webrtc-events.session-invitation:read

:thinking:

---

I have changed the suggestion according to:
https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#apis-which-deal-with-explicit-subscriptions

> Scopes should be represented as below for APIs that offer explicit event subscriptions with action **read** and **delete**, for example:
> 
>   - API Name: device-roaming-subscriptions
>   - Grant-level, action on resource: read, delete
> 
> results in scope: `device-roaming-subscriptions:read`. This type of formulation is not used for the creation action.
> 
> The format to define scopes for explicit subscriptions with action **creation**, includes the event type in its formulation to ensure that consent is managed at the level of subscribed event types.

---

Ok, perfect, `<api_name>:<grant_level_action>` so based on #57, already approved by all team members, the API will be named **webrtc-events**.

So I will go with:

* `webrtc-events:read`

Hope you agree with this little variation. 
As you mention, it applies the same for create, read, and delete
Thanks

---

Fixed at [93a4bcf](https://github.com/camaraproject/WebRTC/pull/53/commits/93a4bcf6f12c8fa4792f9be936cffe28c7e06d5b)